admin/class-boldgrid-backup-admin-settings.php

```diff
@@ -256,20 +256,86 @@ public function get_settings( $raw = false ) {
 		$settings['encrypt_db'] = isset( $settings['encrypt_db'] ) ? (bool) $settings['encrypt_db'] : false;
 
 		// Auto Updates.
+		$settings['auto_update'] = $this->auto_update_settings( $settings );
+
+		// Return the settings array.
+		return $settings;
+	}
+
+	/**
+	 * Get the Auto Update Settings.
+	 *
+	 * With the new changes of wp5.5, there are new changes to the auto update settings.
+	 * Moving this to it's own method allows us to isolate that logic from the get_settings() method.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param array $settings The Settings array.
+	 *
+	 * @return array
+	 */
+	public function auto_update_settings( $settings ) {
+		global $wp_version;
 		if ( empty( $settings['auto_update'] ) ) {
-			$settings['auto_update'] = array(
+			$auto_update_settings = array(
 				'days'    => 0,
 				'plugins' => array(
-					'default' => false,
+					'default' => '0',
 				),
 				'themes'  => array(
-					'default' => false,
+					'default' => '0',
 				),
 			);
+		} else {
+			$auto_update_settings = $settings['auto_update'];
 		}
+		// If version is wp_version is greater than 5.4.99 then it must be 5.5.
+		if ( version_compare( $wp_version, '5.4.99', 'gt' ) ) {
+			$auto_update_plugins = get_option( 'auto_update_plugins', array() );
 
-		// Return the settings array.
-		return $settings;
+			// If the plugin is listed in wp's auto_update_plugin option, then enable it in our settings.
+			foreach ( $auto_update_plugins as $plugin ) {
+				$auto_update_settings['plugins'][ $plugin ] = '1';
+			}
+		}
+
+		return $auto_update_settings;
+	}
+
+	/**
+	 * Updates the WordPress auto_update options.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param array $new_settings New settings.
+	 * @param bool  $is_theme     Whether or not the settings being updated are for a theme.
+	 */
+	public function set_autoupdate_options( $new_settings, $is_theme = false ) {
```

Comment:
The introduction of `$is_theme = false` as a parameter feels like it makes this method rigid and inflexible.

I don't think this is needed:
```
// Depending on whether this is being run for themes or plugins, get the correct option table.
$auto_update_field = get_option( $is_theme ? 'auto_update_themes' : 'auto_update_plugins', array() );
// For each offer ( theme or plugin ) passed to this function.
foreach ( $new_settings[ $is_theme ? 'themes' : 'plugins' ] as $offer => $enabled ) {
```

Our goal isn't to write the fewest amount of lines, it's to write clean and organized code that's maintainable. Trying to decipher what's going on is going to be a pain for the next person to edit this section of code. If something needs to change for plugins but not themes (or vice versa), this method is going to get even more trickier.

Reply:
Would you recommend having this separate as two different methods?